test/custom_resource_crud/custom_resource_crud_c3_test.go

```diff
@@ -0,0 +1,296 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package crcrud
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	enterprisev1 "github.com/splunk/splunk-operator/pkg/apis/enterprise/v1"
+	splcommon "github.com/splunk/splunk-operator/pkg/splunk/common"
+	"github.com/splunk/splunk-operator/test/testenv"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+var _ = Describe("Crcrud test for SVA C3", func() {
+
+	var deployment *testenv.Deployment
+	var defaultCPULimits string
+	var newCPULimits string
+
+	BeforeEach(func() {
+		var err error
+		deployment, err = testenvInstance.NewDeployment(testenv.RandomDNSName(3))
+		Expect(err).To(Succeed(), "Unable to create deployment")
+		defaultCPULimits = "4"
+		newCPULimits = "2"
+
+	})
+
+	AfterEach(func() {
+		// When a test spec failed, skip the teardown so we can troubleshoot.
+		if CurrentGinkgoTestDescription().Failed {
+			testenvInstance.SkipTeardown = true
+		}
+		if deployment != nil {
+			deployment.Teardown()
+		}
+	})
+
+	Context("Clustered deployment (C3 - clustered indexer, search head cluster)", func() {
+		It("crcrud: can deploy indexer and search head cluster, change their CR, update the instances", func() {
+
+			// Deploy Single site Cluster and Search Head Clusters
+			err := deployment.DeploySingleSiteCluster(deployment.GetName(), 3, true /*shc*/)
+			Expect(err).To(Succeed(), "Unable to deploy cluster")
+
+			// Ensure that the Cluster Master goes to Ready phase
+			testenv.ClusterMasterReady(deployment, testenvInstance)
+
+			// Ensure Indexers go to Ready phase
+			testenv.SingleSiteIndexersReady(deployment, testenvInstance)
+
+			// Ensure Search Head Cluster go to Ready phase
+			testenv.SearchHeadClusterReady(deployment, testenvInstance)
+
+			// Verify MC Pod is Ready
+			testenv.MCPodReady(testenvInstance.GetName(), deployment)
+
+			// Verify RF SF is met
+			testenv.VerifyRFSFMet(deployment, testenvInstance)
+
+			// Verify CPU limits on Indexers before updating the CR
+			indexerPodName := fmt.Sprintf(testenv.IndexerPod, deployment.GetName(), 0)
+			testenv.VerifyCPULimits(deployment, testenvInstance.GetName(), indexerPodName, defaultCPULimits)
+			indexerPodName = fmt.Sprintf(testenv.IndexerPod, deployment.GetName(), 1)
+			testenv.VerifyCPULimits(deployment, testenvInstance.GetName(), indexerPodName, defaultCPULimits)
+			indexerPodName = fmt.Sprintf(testenv.IndexerPod, deployment.GetName(), 2)
+			testenv.VerifyCPULimits(deployment, testenvInstance.GetName(), indexerPodName, defaultCPULimits)
+
+			// Change CPU limits to trigger CR update
+			idxc := &enterprisev1.IndexerCluster{}
+			instanceName := fmt.Sprintf("%s-idxc", deployment.GetName())
+			err = deployment.GetInstance(instanceName, idxc)
+			if err != nil {
+				testenvInstance.Log.Error(err, "Unable to fetch Indexer Cluster deployment")
```

Comment:
Don't we need to return here, in case of an error?

Reply:
Thanks @sgontla for pointing this out we should fail the test case here. 
Eg `Expect(err).To(Succeed(), "Unable to get instance of indexer cluster") and we can remove `if` statement